R/period.R

```diff
@@ -402,8 +402,18 @@ yearquarter.Date <- yearquarter.POSIXt
 yearquarter.character <- function(x) {
   if (is_empty(x)) return(as_yearquarter(x))
 
-  anytime::assertDate(x)
-  as_yearquarter(anytime::anydate(x))
+  qtr_regex <- "(?i)(q|qtr|quarter)(?-i) *"
+  if(all(grepl(paste0("\\d{4} *", qtr_regex, "\\d|", qtr_regex, "\\d *\\d{4}"), x, perl = TRUE))){
```

Comment:
Can you refactor the code to extract digits only without using `qtr_regrex` for quarter and year?

Reply:
You can refactor it - using `qtr_regex` for both quarter and year is safer than just matching digits.